cpp/src/arrow/io/file.cc

```diff
@@ -624,16 +624,22 @@ class MemoryMappedFile::MemoryMap : public MutableBuffer {
       is_mutable_ = false;
     }
 
-    void* result = mmap(nullptr, static_cast<size_t>(file_->size()), prot_flags, map_mode,
-                        file_->fd(), 0);
-    if (result == MAP_FAILED) {
-      std::stringstream ss;
-      ss << "Memory mapping file failed, errno: " << errno;
-      return Status::IOError(ss.str());
+    size_ = file_->size();
+
+    void* result = nullptr;
+
+    // Memory mapping fails when file size is 0
```

Comment:
Is it desirable to set `data_` to a dummy non-null result when size is 0? For example a private static 0-length array. Some code may trip on a null pointer.

Reply:
In theory, if the length is zero, this `data_` member should never be used for anything, so I might argue that having an ASAN / valgrind failure for passing nullptr someplace it shouldn't go would be a good thing